modules/cloudsql/main.tf

```diff
@@ -21,39 +21,89 @@ resource "google_project_service" "project_service" {
   disable_on_destroy         = false
 }
 
-resource "google_sql_database_instance" "main" {
-  name             = var.instance_name
-  database_version = "POSTGRES_15"
-  region           = var.region
-  settings {
-    # Second-generation instance tiers are based on the machine
-    # type. See argument reference below.
-    tier = "db-f1-micro"
-  }
-
-  deletion_protection = false
+resource "random_password" "pwd" {
+  length  = 16
+  special = false
 }
 
-resource "google_sql_database" "database" {
-  name     = "pgvector-database"
-  instance = var.instance_name
+// TODO: Migrate to terraform-google-modules/sql-db/google//modules/private_service_access
+// once https://github.com/terraform-google-modules/terraform-google-sql-db/issues/585 is resolved.
+// We define a VPC peering subnet that will be peered with the
+// Cloud SQL instance network. The Cloud SQL instance will
+// have a private IP within the provided range.
+// https://cloud.google.com/vpc/docs/configure-private-services-access
 
-  depends_on = [google_sql_database_instance.main]
+data "google_compute_network" "main" {
+  name    = var.network_name
+  project = var.project_id
 }
 
-resource "random_password" "pwd" {
-  length  = 16
-  special = false
+resource "google_compute_global_address" "google-managed-services-range" {
```

Comment:
Had a discussion with @imreddy13, the ideal to setup PSA is in infrastructure module. Can we move this PSA required resource blocks to infra module? 

And cloudsql module will always expect that given network will have PSA already enabled and configured. (Also mention same in readme and variable description.)

In infrastructure whenever create_network=true, always setup network with PSA access.. for existing network, lets skip setting PSA for phase 1 at least. 

Once phase 1 works good for all and things are pretty stable, then we can support `create_psa` flag that could configure PSA on existing network.

Reply:
> In infrastructure whenever create_network=true, always setup network with PSA access.. for existing network, lets skip setting PSA for phase 1 at least.

Sounds reasonable. Will update.